lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -106,6 +106,34 @@ def provision
       end
 
       @logger.notify 'Spent %.2f seconds grabbing VMs' % (Time.now - start)
+
+      start = Time.now
+      @logger.notify 'Tagging vmpooler VMs'
+
+      tags = {
+        'jenkins_build_url' => @options[:jenkins_build_url],
+        'department' => @options[:department],
+        'project' => @options[:project],
+        'created_by' => @options[:created_by]
+      }
+
+      @hosts.each_with_index do |h, i|
+        uri = URI.parse(@options[:pooling_api] + '/vm/' + h['vmhostname'].split('.')[0])
+
+        http = Net::HTTP.new(uri.host, uri.port)
+        request = Net::HTTP::Put.new(uri.request_uri)
```

Comment:
Should probably put this in a begin/rescue for HTTP errors, in case the vpooler goes down.  Would want to raise reasonable errors.